### ISSUE TITLE
Position of the lightbox goes off of the screen when height value is set in px

### ISSUE THREAD

**Issue by robbiegod** [biati-digital/glightbox#275]
**Describe the bug**
When the Glightbox is opened the lightbox is half way off of the screen from the top of the screen OR it is halfway off of the screen at the bottom.  Issue is getting the lightbox centered when a px height is entered.

I have a button on my website that launches a Glightbox. When I set "height: auto" on the "ginner-container" class div, then the position is correct.  It seems the script is adding a height: 960px; to the ginner-container.  If I removed the height in my settings then I get a lightbox that is 506px (the default height) and the position is off of the screen at the bottom.

**Are you able to reproduce the bug in the demo site**
Yes and No.  I experimented by adding height:auto !important; to the ginner-container in my standardSlightHTML custom template and that seemed to fix it.  I removed that line and it still seems to be working. I don't know what is going on now.  Could it be an issue that the script isn't fully loading and only after so many refreshes it is finally in my cache.

Have you ever seen this with the position of the lightbox?

**To Reproduce**
1. Go to my dev site: https://cm-sandbox.crosbydev.net/our-work/case-studies/military-onesource/
2. Scroll down to the module with the text "Making moving an adventure", click on the View Gallery button.
3. What result do you get?

**Expected behavior**
The modal opens, centered in the middle of the page. The lightbox should be 1280px by 960px

**Post the code you are using**

```
let standardSlideHTML = `<div class="gslide">
   <div class="gslide-inner-content">
       <div class="ginner-container">
           <div class="gslide-media embed-responsive embed-responsive-4by3">
           </div>
           <div class="gslide-description">
               <div class="gdesc-inner">
                   <h4 class="gslide-title"></h4>
                   <div class="gslide-desc"></div>
               </div>
           </div>
       </div>
   </div>
</div>`;

   var standardGalleryIframes = GLightbox({
    selector: '.standard-lightbox',
    width: '1280px',
    height: '960px',
    skin: 'modern',
    closeOnOutsideClick: true,
    slideHTML: standardSlideHTML
  });
```

My button looks like this: 

`<a href="/projects/military-onesource-mastering-your-pcs/" class="standard-lightbox">`

**Screenshots**
Sorry, i am now having problems replicating this issue, but I had seen this before on an other page and i thought i had fixed it. I am just going through the website and replacing fancybox with GLightbox.

**Desktop:**
 - OS: Windows 10
 - Browser: Chrome
 - Version:  92.0.4515.131

**Comment by frankie-tech** [comment]
Hey there @robbiegod,

Unless I'm reading this wrong, it sounds like you're not able to consistently reproduce this bug.

It *sounds* like this is just a caching issue, but I really have no way of knowing.

If you can recreate the bug in a CodePen or codesandbox without custom HTML, we can look into this further.

**Comment by robbiegod** [comment]
@frankie-tech Maybe you can help me with something else then that is sort of related to this.

The size of my lightbox is 1280px by 960px.  I know the lightbox is supposed to be responsive, but it seems to only be responsive as it relates to width and not height.  So, if I move my window down to 1366 x 768, the box is cut off at the top and the bottom.  It doesn't scale to fit the height.

I am wondering if you have plans or if there is a setting I can add that will make the lightbox scale by both width and height, then it will fit on all shapes and sizes of the browsers.

For now i added a css media query that brings the width down to 960px, but I'd rather take that out and just have it scale down the height of the browser changes.

**Comment by frankie-tech** [comment]
@robbiegod 

Sorry, we don't offer support outside of bugs and feature requests.

There aren't plans to add a feature like that.

You could probably use some viewport units to solve it with CSS.

**Comment by robbiegod** [comment]
@frankie-tech This is a feature request then 💯 

The feature would be to make the modal window that is of type iframe fully responsive. Thank you!

**Comment by frankie-tech** [comment]
I think I may have misled you with my last comment.

There's plenty of resources out there on how to code a responsive iframe, since they are historically a pain to style. You can look at stuff like FitVids.js or the infamous aspect ratio padding hack.

This [Smashing Magazine](https://www.smashingmagazine.com/2014/02/making-embedded-content-work-in-responsive-design/#the-markup-for-embedded-content) article might be a good starting point for you too.

For now though, I'm going to close this issue since it is out of scope for GLightbox.

**Comment by robbiegod** [comment]
@frankie-tech Thanks for taking a look and for giving me some options, much appreciated.

Just so I understand, can I submit feature requests here?  I'm assuming that since you are denying my request that I can't or you are not interested in fixing the issue I am having.

Maybe this should be a bug then? Right now, if my lightbox is 1280 x 960, the lightbox doesn't work out of the box at certain resolutions like 1366x768 - which is still widely used.  The lightbox goes off the top and bottom of the screen, its just too tall to see the whole thing.  The lightbox iframe does not scale to the height of the screen. 

I'm not trying to be a pain or anything, I was trying to be helpful to you (and to myself).  

For now, I setup a media query that just changes the width of the box so its small enough to fit.  The solution is not great, but it'll do for now.  I'll look at the articles you referred me too (thanks!).

**Comment by frankie-tech** [comment]
Don't worry you're not being a pain.

The issue you had originally pointed out, as I understand it, isn't created by GLightbox. I'll review the CSS code again, but it sounds like it is just due to the hellish nature of styling iframe elements.

If you can produce a codepen or something that can demonstrate the bug, then we can look more into it.

But without that, all I've got is what you've written down here. Some of the questions in the original issue, your answers aren't very clear

> **Are you able to reproduce the bug in the demo site**
> Yes and No.

> **Screenshots**
> Sorry, i am now having problems replicating this issue

It's hard to find a bug with little proof that it exists other than "I got it to show up, but it's not happening consistently."

If we're going to treat this as a feature request though, I'd appreciate if you could submit an issue using the feature request template since the topic of this thread has hopped around a lot.

It will also give us, and anyone else who may want to give it a shot, a better idea of what is expected.
